tests/framework/extensions/clusters/import.go

```diff
@@ -0,0 +1,185 @@
+package clusters
+
+import (
+	"context"
+	"time"
+
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/tests/framework/clients/dynamic"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	ext_unstructured "github.com/rancher/rancher/tests/framework/extensions/unstructured"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/rest"
+)
+
+// kubeConfig is a basic kubeconfig that uses the pod's service account
+const kubeConfig = `
+apiVersion: v1
+kind: Config
+clusters:
+- name: cluster
+  cluster:
+    certificate-authority: /run/secrets/kubernetes.io/serviceaccount/ca.crt
+    server: https://kubernetes.default
+contexts:
+- name: default
+  context:
+    cluster: cluster
+    user: user
+current-context: default
+users:
+- name: user
+  user:
+    tokenFile: /run/secrets/kubernetes.io/serviceaccount/token
+`
+
+// ImportCluster creates a job using the given rest config that applies the import yaml from the given management cluster.
+func ImportCluster(client *rancher.Client, cluster *management.Cluster, rest *rest.Config) error {
+	// create a sub session to clean up after we apply the manifest
```

Comment:
I'm just curious why you are using the management/v3 cluster rather than the provisioning/v1 cluster.